### PR TITLE
Add SynthResult to the API

### PR DESCRIPTION
**Pull request by ajreynol**
Does not modify the code to return a SynthResult yet, just adds the class.

Depends on #8352.

### CODE REVIEW

**Comment by aniemetz** [comment]
I have pushed a commit adding missing docs files and fixing references to functions.

File: src/api/java/io/github/cvc5/api/Solver.java [hunk 1]
```diff
@@ -2699,6 +2699,17 @@ public Result getNullResult()
 
   private native long getNullResult(long pointer);
 
+  /**
+   * @return null synth result
+   */
+  public SynthResult getNullSynthResult()
```
[the remark this answers is not in the document]
Comment (ajreynol) [reply src/api/java/io/github/cvc5/api/Solver.java:2705]: I'm not sure whether we need it, but I guess it mirrors C++ where you can create a null SynthResult without a solver, whereas in the Java API I suppose you always need access to a solver?

[the remark this answers is not in the document]
Comment (aniemetz) [reply src/api/java/io/github/cvc5/api/Solver.java:2705]: Ah, that would make sense, yes.

[the remark this answers is not in the document]
Comment (mudathirmahgoub) [reply src/api/java/io/github/cvc5/api/Solver.java:2705]: Yes. These getNull `Term, Sort, Result` mirrors the unit tests. Probably we would not need them if we remove the dependency on Solver.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**